app/components/publications/batch-documents-publication-modal.hbs

```diff
@@ -0,0 +1,49 @@
+<WebComponents::AuModal>
+  <WebComponents::AuModal::Header @title={{ t "documents" }} @closeModal={{this.onCancel}} />
+  <WebComponents::AuModal::Body>
+    {{!-- template-lint-disable table-groups --}}
+    <table class="auk-table auk-table--styled auk-table--bordered">
+      <thead>
+        <tr>
+          <th>{{t "name"}}</th>
+          <th>{{t "document-type"}}</th>
+          <th></th>
+        </tr>
+      </thead>
+
+      <Documents::DocumentList
```

Comment:
Ik denk dat het hier beter was geweest dat de DocumentList.groupDocumentsByContainer-methode los stond van de rendering van een div. Het lijkt me geen taak van een component. Hierdoor moest ik een workaround voorzien.

Reply:
Je hebt gelijk dat de `<div>` in de template de component minder generiek bruikbaar maakt. Het is een "beperking" van de [ember-render-modifier](https://github.com/emberjs/ember-render-modifiers) dat die een DOM element nodig heeft om op in te haken, vandaar de `<div>` in de template.

Ik denk dat we voor deze case de [ember-element-helper addon](https://github.com/tildeio/ember-element-helper) kunnen gebruiken. Die laat toe om de `tagName` van de `Documents::DocumentList` component dynamisch te maken. Gebasseerd op [RFC #389](https://github.com/emberjs/rfcs/pull/620) gaat die op termijn ook in Ember landen. 

Ik zou de interface van `Documents::DocumentList` uitbreiden zodanig dat een `@tagName` meegegeven kan worden, die default `div` is.

Reply:
Is dat geen nodeloze workaround? Als ik van de methode groupDocumentsByContainer een static methode maak (vb. op Piece), dan heb ik die component niet nodig. did-update lijkt me hier toch niet nodig. Ik vermoed dat die er enkel is voor het geval documenten toegevoegd worden.

Reply:
Ik vind dit geen workaround: het aanpassen van de tag name is effectief wat we willen.

De `Documents::DocumentList` is een component die op nog andere plaatsen in de applicatie gebruikt wordt waar wel  documenten toegevoegd kunnen worden. In die gevallen is de `did-update` wel noodzakelijk.

Het groeperen van de documenten per document container is net de functionaliteit die deze component biedt. Als je die eruit haalt, heeft de component geen meerwaarde meer. Of begrijp ik je verkeerd?

Reply:
Ik vermoed dat Marnik hier doelt op het concept "renderless component". Dat is een patroon dat je niet overal tegenkomt, maar in ember wel breed gebruikt. Het is dus geen workaround.

Reply:
OK. Heb deze addon toegevoegd.